docs/docs/architecture/security.md

```diff
@@ -0,0 +1,105 @@
+The goal here is to implement a  [cross-signed certificate chain](https://www.ssltrust.com.au/blog/understanding-certificate-cross-signing).
+In that way, the Marbles see the *Marble Root Certificate* as a self-signed root certificate. Hence, they're dealing with a terminating certificate chain without knowing about the Coordinator's *Root CA Certificate*.
+The "outside world" sees an intermediate certificate signed by the Coordinator's *Root CA Certificate*.
+The Coordinator generates a unique leaf *Marble Certificate* and corresponding key pair using ECDSA with P256 for every Marble.
+The *Marble Root Certificate* signs the *Marble Certificate*.
+The *Marble Certificate* is provisioned to the Marble's enclave via the secure channel established during the [attestation procedure](../features/attestation.md).
+Depending on the Marble's runtime, the certificate can be used [manually](../workflows/add-service.md#make-your-service-use-the-provided-tls-credentials) or [automatically](../features/transparent-TLS.md) to establish mutually authenticated TLS connections.
+
+![PKI Certificate chain](../_media/cert-chain.svg)
+
+
+## Attested TLS (aTLS)
+
+In a CC environment, attested TLS (aTLS) can establish secure connections between two parties using the remote attestation features of the CC components.
+aTLS modifies the TLS handshake by embedding an attestation statement into the TLS certificate. Instead of relying on a certificate authority, aTLS uses this attestation statement to establish trust in the certificate.
+The protocol can be used by clients to verify a server certificate, by a server to verify a client certificate, or for mutual verification (mutual aTLS).
+
+
+## Encryption of state
+
+The Coordinator holds the MarbleRuns state, which consists of the [manifest](../features/manifest.md), the [managed secrets](../features/secrets-management.md), and the [certificates for its CA](../features/attestation.md).
+The state is stored encrypted in persistent storage. Therefore Constellation uses [AES128-GCM](https://www.rfc-editor.org/rfc/rfc5116#section-5.1) and a generated 16-byte Data Encryption Key (KEK) to seal the state to persistent storage.
+The DEK is also sealed to persistent storage to recover the state in case of a restart autonomously.
+[SGX sealing](https://www.intel.com/content/www/us/en/developer/articles/technical/introduction-to-intel-sgx-sealing.html) is used for that purpose.
+The Coordinator encrypts the DEK with a key encryption key (KEK).
+The KEK MarbleRun uses the SGX sealing key called `Product key`, which is bound to its `Product ID` and the enclave's author `MRSIGNER` identity.
+In other words, a fresh and benign enclave instance of the same identity can recover that key.
+Hence, as long as the Coordinator is restarted on the same CPU, it can obtain the same KEK from the CPU based on its identity, decrypt the DEK and recover its state.
```

Comment:
```suggestion
Hence, if the Coordinator is restarted on the same CPU, it can obtain the same KEK from the CPU, decrypt the DEK, and recover its state.
```
suggest shortening this